server/src/main/java/org/elasticsearch/search/internal/ContextIndexSearcher.java

```diff
@@ -79,6 +79,17 @@ public ContextIndexSearcher(IndexReader reader, Similarity similarity, QueryCach
     }
 
     public void setProfiler(QueryProfiler profiler) {
+        if (profiler != null) {
```

Comment:
This change seems not to be related with the changes in this PR?

Reply:
Right, I'll remove, thanks.

Reply:
Fixed in https://github.com/elastic/elasticsearch/pull/48399/commits/95820b1a0bc6ee576ffce19bbf440c90ecb8a7a2